src/StopWatch.tsx

```diff
@@ -1,7 +1,105 @@
-import React from 'react'
+/**
+ * @fileoverview module component that hosts the main stop watch functionality
+ */
 
-export default function StopWatch() {
-    return(
-        <div></div>
-    )
-}
\ No newline at end of file
+import React from "react";
+import { StopWatchButton } from "./components";
+import { Time } from "./components";
+
+// global variable to store the number of laps recorded
+let lapTimeId = 0;
+
+const StopWatch: React.FC = () => {
+  // not specifying the type here for useState since typescript can infer this from the initial state value
+  const [time, setTime] = React.useState(0);
+  const [isTimerRunning, setTimerRunning] = React.useState(false);
+  // state array to store the recorded lap times
+  const [lapTimes, setLapTimes] = React.useState<
+    { id: number; time: number }[]
```

Comment:
We could extract and export this type as a separate type when the type is used in multiple places. For now, left it as it is since it is only used in one place.